CMake/resolve_dependency_modules/simdjson.cmake

```diff
@@ -27,6 +27,16 @@ message(STATUS "Building simdjson from source")
 FetchContent_Declare(
   simdjson
   URL ${VELOX_SIMDJSON_SOURCE_URL}
-  URL_HASH ${VELOX_SIMDJSON_BUILD_SHA256_CHECKSUM})
+  URL_HASH ${VELOX_SIMDJSON_BUILD_SHA256_CHECKSUM}
+  SOURCE_DIR ${CMAKE_BINARY_DIR}/_deps/simdjson-src/simdjson)
 
-FetchContent_MakeAvailable(simdjson)
+# When simdjson is built with cmake it places simdjson/singleheader/simdjson.h
```

Comment:
Cc: @kgpai @assignUser to make sure this doesn't clash with any of the rework being done in the cmake files.